pkg/cache/discovery_cache.go

```diff
@@ -61,8 +98,24 @@ func updateDiscoveryCache(discoveryClient *discovery.DiscoveryClient) error {
 
 			discoveryCache.Set(fmt.Sprintf("%s/%s", gv.String(), resource.Kind), true)
 			discoveryCache.Set(gv.String(), true)
+
+			checkAndUpdateServiceMesh(gv.Group)
```

Comment:
this logic can technically overwrite the value in ways we don't want, we'd ideally want istio to take priority over linkerd over cilium

Reply:
oh yeah, forgot about adding priority list

Reply:
Done